File: app/src/main/java/com/example/miaula/Controllers/CourseController.java
package com.example.miaula.Controllers;

import android.content.SharedPreferences;

import com.example.miaula.Models.Course;
import com.example.miaula.Models.Student;

import java.util.ArrayList;

public class CourseController {

    private static CourseController instance;
    private ArrayList<Course> courses;

    public static CourseController getInstance(){
        if (instance == null){
            instance = new CourseController();
        }
        return instance;
    }

    private CourseController() {

    }

    public ArrayList<Course> getCourses() {
        return courses;
    }

    public void setCourses(ArrayList<Course> courses) {
        this.courses = courses;
    }

    public Course getCourse(int idCourse){
        for (int x=0; x<courses.size(); x++){
            if (this.courses.get(x).getIdCourse() == idCourse)
                return this.courses.get(x);
        }
        return null;
    }

    public void addCourse(Course course) {
        this.courses.add(course);
    }

    public void addStudent(int idCourse,Student student) {
        Course course = new Course();
        for (int x=0; x<courses.size(); x++){
            if (this.courses.get(x).getIdCourse() == idCourse)
                course = this.courses.get(x);;
                course.setStudent(student);
            System.out.println("Course..:"+ course);
        }
    }

}

File: app/src/main/java/com/example/miaula/Controllers/CourseController.kt
package com.example.miaula.Controllers

import com.example.miaula.models.Course

class CourseController private constructor() {
    var courses: ArrayList<Course>? = null
    fun getCourse(idCourse: Int): Course? {
        for (x in courses!!.indices) {
            if (courses!![x].idCourse == idCourse) return courses!![x]
        }
        return null
    }

    fun addCourse(course: Course) {
        courses!!.add(course)
    } //    public void addStudent(int idCourse,Student student) {

    //        Course course = new Course();
    //        for (int x=0; x<courses.size(); x++){
    //            if (this.courses.get(x).getIdCourse() == idCourse)
    //                course = this.courses.get(x);;
    //                course.setStudent(student);
    //            System.out.println("Course..:"+ course);
    //        }
    //    }
    companion object {
        var instance: CourseController? = null
            get() {
                if (field == null) {
                    field = CourseController()
                }
                return field
            }
            private set
    }
}